### PR TITLE
Baseplate.spec calls for `<protocol>_client_name` instead of `<protocol>_slug`

### PR DESCRIPTION
**Pull request by nanassito**
## 💸 TL;DR
The Baseplate.spec calls for the client pools to have a label in the form `<protocol>_client_name` but baseplate.go uses `<protocol>_slug`. This PR addresses this discrepancy.

## 🧪 Testing Steps / Validation
Updated unit tests

## ✅ Checks

- [ ] CI tests (if present) are passing
- [ ] Adheres to code style for repo
- [ ] Contributor License Agreement (CLA) completed if not a Reddit employee

### CODE REVIEW

**Review by kylelemons — APPROVED** [review]
(I would probably rename the constants too, but don't feel strongly)

**Review by konradreiche — APPROVED** [review]
:snail:

**Comment by nanassito** [comment]
I don't have permission to merge in this repo. Can one of you do that for me please ?

File: grpcbp/prometheus.go [hunk 1]
```diff
@@ -17,7 +17,8 @@ const (
 	typeLabel       = "grpc_type"
 	successLabel    = "grpc_success"
 	codeLabel       = "grpc_code"
-	serverSlugLabel = "grpc_slug"
+	serverSlugLabel = "grpc_slug" // Deprecated, will be removed after 2022-09-01
```
Comment (mterwill) [inline grpcbp/prometheus.go:20]: Shame there's not a better way to surface this 😕

File: thriftbp/client_middlewares_test.go [hunk 2]
```diff
@@ -467,12 +467,13 @@ const (
...
-	exceptionLabel           = "thrift_exception_type"
-	baseplateStatusLabel     = "thrift_baseplate_status"
-	baseplateStatusCodeLabel = "thrift_baseplate_status_code"
-	remoteServiceSlugLabel   = "thrift_slug"
+	methodLabel                  = "thrift_method"
+	successLabel                 = "thrift_success"
+	exceptionLabel               = "thrift_exception_type"
+	baseplateStatusLabel         = "thrift_baseplate_status"
+	baseplateStatusCodeLabel     = "thrift_baseplate_status_code"
+	remoteServiceSlugLabel       = "thrift_slug"
```
Comment (mterwill) [inline thriftbp/client_middlewares_test.go:475]:
```suggestion
	remoteServiceSlugLabel       = "thrift_slug" // Deprecated, will be removed after 2022-09-01
```

**Review by mterwill — APPROVED** [review]

**Pull request merged**